wgpu-core/src/device/global.rs

```diff
@@ -1767,6 +1767,135 @@ impl Global {
         (id, Some(error))
     }
 
+    #[cfg(feature = "wgsl")]
+    pub(crate) fn device_get_or_create_dispatch_indirect_validation_pipeline<A: HalApi>(
+        &self,
+        device_id: DeviceId,
+    ) -> Result<
+        (id::ComputePipelineId, id::BindGroupLayoutId),
```

Comment:
Could we wrap this functionality in an encapsulated type with constructor and execute functions, preferably in its own file?